### PR TITLE
Add button to download logs from zwave_js logs page

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Adds a new download button to the zwave_js log page that lets a user download the logs shown in the frontend (used automation trace downloads as a reference). I know nothing about CSS and HTML so the placement of this icon may not be great, open to suggestions

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts [hunk 1]
```diff
@@ -114,6 +128,15 @@ class ZWaveJSLogs extends SubscribeMixin(LitElement) {
     );
   }
 
+  private _downloadLogs() {
+    const aEl = document.createElement("a");
+    aEl.download = `zwave_js ${this.startDateTime.toISOString()} to ${new Date().toISOString()}.log`;
```
Comment (cgarwood) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:133]: Avoid spaces in filenames

Comment (cgarwood) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:133]: 2 ISO date strings is going to generate a really long filename. I'm wondering if the timestamps are actually needed or if we should just use something like `zwave_js.log` let the users rename them post download or during the download (depending on if their browser is configured to prompt for filename/download location)

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts [hunk 2]
```diff
@@ -92,6 +93,17 @@ class ZWaveJSLogs extends SubscribeMixin(LitElement) {
...
+              <mwc-icon-button
+                label="Download Logs"
+                @click=${this._downloadLogs}
+              >
+                <ha-svg-icon
+                  .title="Download Logs"
+                  .path=${mdiDownload}
+                ></ha-svg-icon>
+              </mwc-icon-button>
+            </div>
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:106]:
```suggestion
              <mwc-icon-button
                label="Download Logs"
                @click=${this._downloadLogs}
              >
                <ha-svg-icon
                  .path=${mdiDownload}
                ></ha-svg-icon>
              </mwc-icon-button>
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts [hunk 3]
```diff
@@ -92,6 +93,17 @@ class ZWaveJSLogs extends SubscribeMixin(LitElement) {
                   `
                 : ""}
             </div>
+            <div>
+              <mwc-icon-button
+                label="Download Logs"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:98]: This should be localized

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts [hunk 4]
```diff
@@ -114,6 +126,15 @@ class ZWaveJSLogs extends SubscribeMixin(LitElement) {
     );
   }
 
+  private _downloadLogs() {
+    const aEl = document.createElement("a");
+    aEl.download = `zwave_js.log`;
+    aEl.href = `data:text/plain;charset=utf-8,${encodeURI(
+      this._textarea!.value
+    )}`;
+    aEl.click();
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:135]: Use `fileDownload`: <https://github.com/home-assistant/frontend/blob/dev/src/util/file_download.ts>

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts [hunk 5]
```diff
@@ -92,6 +94,14 @@ class ZWaveJSLogs extends SubscribeMixin(LitElement) {
                   `
                 : ""}
             </div>
+            <mwc-icon-button
+              label=${this.hass.localize(
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts:98]:
```suggestion
              .label=${this.hass.localize(
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**